libbpf-tools/opensnoop.c

```diff
@@ -76,10 +79,22 @@ static const struct argp_option opts[] = {
 	{},
 };
 
+static int parse_uid(const char *uid_arg, uid_t *uid) {
```

Comment:
why extracting into a function? it's called from just one place in code

Reply:
It's basically same reason, as for `utils.h`. Parsing `uid` is something that done by multiple tools, so I thought that it's something that could be moved into separate file later (maybe when there will be more tools that will require this).

Reply:
Yeah, let's think about extracting reusable code (if it makes sense) once we have few use cases. Having extra headers with trivial functions reduces readability without really bringing much benefit (apart from saving 3 lines of trivial code). I'd also be wary about sharing code between BPF and user-space. It might work in simple cases, but could be subtly broken in others.